Fix DwUsbHostDxe to allow for use with USB NICs

mcb30 opened the pull request.

pbatard commented:
Looks good to me.

There are just a few issues with tabs, but I'll fix that.

For the record, the EDK2 has a Python script (`PatchCheck.py`) that can check preliminary stuff for submission (such as use of tabs, lines that are too long, etc.). For instance, if you applied that above 3 patches and want to validate them, you can run:
```
root@vm-debian9 /usr/src/edk2-platforms # python /usr/src/edk2/BaseTools/Scripts/PatchCheck.py -3
Checking git commit: d8d109ec3f
The commit message format passed all checks.
Code format is not valid:
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:        DEBUG ((DEBUG_ERROR, "Channel %u did not halt\n", Channel));

Checking git commit: a0805e7050
The commit message format passed all checks.
Code format is not valid:
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:                              EFI_TIMER_PERIOD_MILLISECONDS (1));
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:        break;
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:                         DWC2_HCINT_CHHLTD, 1);
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:        Status = EFI_DEVICE_ERROR;
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:        Status = EFI_TIMEOUT;

Checking git commit: 128b7a87fb
The commit message format passed all checks.
Code format is not valid:
 * Tab character used
   File: Platform/RaspberryPi/RPi3/Drivers/DwUsbHostDxe/DwUsbHostDxe.c
   Line:                     DWC2_HCINT_CHHLTD, 1);
``` 

Also for the sake of being overtly explicit about the fact that it's the success condition of waiting for the timeout bit that tells if we should return a timeout status or a different error, I'm probably going to modify the second patch @331 with the following before submitting it:
```C
      if (Status == EFI_SUCCESS) {
	Status = EFI_TIMEOUT;
      } else {
	Status = EFI_DEVICE_ERROR;
      }
```

I'll try to test so more and then I'll submit the patches to the EDK2 mailing list some time this week.

mcb30 committed:
Platform/RPi3: Accept "ethernet" or "ethernet0" aliases in device tree

Older device trees tend to use the alias "ethernet".  Newer device
trees tend to use "ethernet0" since older versions of U-Boot would
skip aliases that do not include an index number.  See, for example,
Linux kernel commit 10b6c0c ("ARM: dts: bcm2835: add index to the
ethernet alias") and U-Boot commit f8e57c6 ("fdt_support: Fixup
'ethernet' aliases not ending in digits").

Accept either "ethernet" or "ethernet0" as being the alias to the node
for which the MAC address should be updated.

Signed-off-by: Michael Brown <mbrown@fensystems.co.uk>
